### DISCUSSION TITLE
Squash PRs before merging

### DISCUSSION THREAD

**Discussion by boldtrn** [maplibre/maplibre-navigation-ios#78] [General]
I think it's great that we see so meany contributions in the repo right now 🎉. I just wanted to start a discussion about squashing PRs before merging them. The current commit history on the main branch is quite clutered as the last few PRs have not been squashed, so I wanted to raise awareness for this. 

As a contributor you can squash your PR before asking for a review.
As a maintainer you can use the merge button in Github to squash the commits.

This will help dramatically to keep the commit history clean.

**Comment by hactar** [comment]
I prefer squash and merge too - why not set it up to be the only available option? I do this for all my projects in the github settings.

**Reply by boldtrn** [reply]
I updated the settings, so from now on, only squashes should be allowed, if not please let me know.

**Reply by hactar** [reply]
<img width="941" alt="Screenshot 2024-07-02 at 15 52 23" src="https://github.com/maplibre/maplibre-navigation-ios/assets/645674/bc40ce59-1963-4743-b5c1-b913bd64dd8b">

Works for me, thanks ☺️

**Reply by boldtrn** [reply]
Ok, let's close this discussion then 👍

**Reply by wipfli** [reply]
The preferred merging in the MapLibre org is squash merging. I am a bit surprised that other methods were still available since the tick was made here in the onboarding issue:

![image](https://github.com/maplibre/maplibre-navigation-ios/assets/53421382/0e08420d-2403-4ff7-84a4-2725ae9dd5bb)

https://github.com/maplibre/maplibre/issues/171

But good that you fixed it now...
